### PR TITLE
Clang-format check

### PR DESCRIPTION
**Pull request by fchinu**
This PR adds an automatic check for the clang-formatting of cpp files. We can consider on requiring to pass this check to be able to merge, but we can discuss this

### CODE REVIEW

File: Clustering/include/cluster.h [hunk 1]
```diff
@@ -1,14 +1,14 @@
-#ifndef CLUSTER_H
-#define CLUSTER_H
+#ifndef CLUSTERING_INCLUDE_CLUSTER_H_
+#define CLUSTERING_INCLUDE_CLUSTER_H_
 
+class Cluster {
+  int fX;
+  int fY;
+  int ClusterSize;
```
Comment (Roberta-Ferioli) [inline Clustering/include/cluster.h:10]:
```suggestion

uint8_t ArrayShape[8];
```

File: Clustering/include/cluster.h [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-class Cluster
-{
-int fX;    
-
-        
-
+  int GetX() const;
+  int GetY() const;
+  int GetClusterSize() const;
 };
```
Comment (Roberta-Ferioli) [inline Clustering/include/cluster.h:15]:
```suggestion
uint8_t GetArrayShape() const;
};

```

**Review by Roberta-Ferioli — CHANGES_REQUESTED** [review]
Thanks @fchinu, can you add the attribute and the function that i suggested at class "Cluster"? They will allow us to represent the cluster shape

**Comment by fchinu** [comment]
HI @Roberta-Ferioli, thanks a lot for your suggestion. I have implemented it. However, I would not change much in the Cluster header file in this PR because it was meant for adding the check for the code formatting. I will add the full clusterer soon!

**Review by Roberta-Ferioli — APPROVED** [review]

**Pull request merged**